### PR TITLE
Preventing service providers from registering if they introduce types…

### PR DESCRIPTION
**Pull request by kmoore-intuit**
… with different field definitions

# What Changed

# Why

Todo:

- [ ] Add tests
- [ ] Add docs

### CODE REVIEW

File: src/main/java/com/intuit/graphql/orchestrator/schema/transform/FederationTransformerPostMerge.java [hunk 1]
```diff
@@ -124,8 +125,11 @@ private void pruneInaccessibleInfo(UnifiedXtextGraph xtextGraph, Collection<Type
             xtextGraph.getTypes().remove(typeDefinition.getName());
             xtextGraph.getBlacklistedTypes().add(typeDefinition.getName());
         } else {
-            getFieldDefinitions(typeDefinition, true)
+            boolean containsInaccessibleFields = getFieldDefinitions(typeDefinition, true)
```
Comment (ashpak-shaikh) [inline src/main/java/com/intuit/graphql/orchestrator/schema/transform/FederationTransformerPostMerge.java:128]: not sure if i understand this part?

Comment (ashpak-shaikh) [reply src/main/java/com/intuit/graphql/orchestrator/schema/transform/FederationTransformerPostMerge.java:128]: Is there a test related to this ?

File: src/test/groovy/com/intuit/graphql/orchestrator/integration/NestedLevelStitchingSpec.groovy [hunk 2]
```diff
@@ -142,5 +177,16 @@ class NestedLevelStitchingSpec extends BaseIntegrationTestSpecification {
...
+        given:
+        topLevelDeprecatedService1 = createSimpleMockService("test1", conflictingSchema, new HashMap<String, Object>())
+        topLevelDeprecatedService2 = createSimpleMockService("test2", conflictingSchema2, new HashMap<String, Object>())
+
+        when:
+        specUnderTest = createGraphQLOrchestrator([topLevelDeprecatedService1, topLevelDeprecatedService2])
 
+        then:
+        def exception = thrown(StitchingException)
+        exception.message == "[test2,test1] are reusing object type NestedType with different field definitions."
```
Comment (kmoore-intuit) [inline src/test/groovy/com/intuit/graphql/orchestrator/integration/NestedLevelStitchingSpec.groovy:190]: change to:
FORBIDDEN: Subgraphs [test2,test1] are reusing type NestedType with different field definitions.

File: src/main/java/com/intuit/graphql/orchestrator/utils/XtextToGraphQLJavaVisitor.java [hunk 3]
```diff
@@ -83,8 +84,10 @@ public class XtextToGraphQLJavaVisitor extends GraphQLSwitch<GraphQLSchemaElemen
   private final Map<String, GraphQLType> graphQLObjectTypes;
   public final Map<String, GraphQLDirective> directiveDefinitions;
   public final HashSet<String> blacklistedTypes;
+  public final Set<String> typesWithInaccessibleFields;
 
   private static final GraphQLScalarType FIELD_SET_SCALAR;
+  private static final String REUSED_CONFLICTING_TYPE = "FORBIDDEN: Subgraphs %s are reusing type %s with different field definitions.";
```
Comment (CNAChino) [inline src/main/java/com/intuit/graphql/orchestrator/utils/XtextToGraphQLJavaVisitor.java:90]: ERRMSG_REUSED_CONFLICTING_TYPE

**Review by CNAChino — APPROVED** [review]

**Review by ashpak-shaikh — APPROVED** [review]

**Pull request merged**